### PR TITLE
Support a user-defined function for `--path-rule-doc` option

### LINKED ISSUE #501: Make the `path-rule-doc` parameter can accept a function
In https://github.com/eslint-react/eslint-react, different categories of eslint rules are organized under different packages. Support for passing a function will greatly improve flexibility, and can help us remove this [patch](https://github.com/eslint-react/eslint-react/blob/main/patches/eslint-doc-generator%401.5.4.patch).

If you agree, I'd be happy to implement this feature.

### PR DESCRIPTION
**Pull request by hyoban**
close #501 

If I missed or messed up something, please let me know and I will correct it ASAP.

### CODE REVIEW

File: test/lib/generate/file-paths-test.ts [hunk 1]
```diff
@@ -267,6 +267,15 @@ describe('generate (file paths)', function () {
       expect(readFileSync('rules/list.md', 'utf8')).toMatchSnapshot();
       expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
     });
+
+    it('generates the documentation using a function for pathRuleDoc', async function () {
+      await generate('.', {
+        pathRuleDoc: (ruleName) => join('rules', ruleName, `${ruleName}.md`),
+        pathRuleList: join('rules', 'list.md'),
```
Comment (bmish) [inline test/lib/generate/file-paths-test.ts]: Let's remove `pathRuleList` from this test as it's not relevant.

File: README.md [hunk 2]
```diff
@@ -182,7 +182,7 @@ There's also a `postprocess` option that's only available via a [config file](#c
 | `--ignore-config` | Config to ignore from being displayed. Often used for an `all` config. Option can be repeated. | |
 | `--ignore-deprecated-rules` | Whether to ignore deprecated rules from being checked, displayed, or updated. | `false` |
 | `--init-rule-docs` | Whether to create rule doc files if they don't yet exist. | `false` |
-| `--path-rule-doc` | Path to markdown file for each rule doc. Use `{name}` placeholder for the rule name. | `docs/rules/{name}.md` |
+| `--path-rule-doc` | Path to markdown file for each rule doc. Use `{name}` placeholder for the rule name. A function can also be provided for this option via a [config file](#configuration-file). | `docs/rules/{name}.md` |
```
Comment (bmish) [inline README.md:185]: Can you add an example of the function to the "Configuration file" section in the readme, similar to how we have examples of the other option functions in that section?

File: test/lib/generate/file-paths-test.ts [hunk 3]
```diff
@@ -267,6 +269,13 @@ describe('generate (file paths)', function () {
       expect(readFileSync('rules/list.md', 'utf8')).toMatchSnapshot();
       expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
     });
+
+    it('generates the documentation using a function for pathRuleDoc', async function () {
+      await generate('.', {
+        pathRuleDoc: (ruleName) => join('rules', ruleName, `${ruleName}.md`),
+      });
+      expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
```
Comment (bmish) [inline test/lib/generate/file-paths-test.ts:277]: I think we still want to snapshot test the README so we can make sure it links to the rule docs properly.

```suggestion
      expect(readFileSync('README.md', 'utf8')).toMatchSnapshot();
      expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
```

**Review by bmish — APPROVED** [review]
Thanks! Great work on this.

**Comment by hyoban** [comment]
> Thanks! Great work on this.

Thank you very much for your suggestions too

**Pull request merged**